src/DeriveHasField.hs

```diff
@@ -63,3 +67,8 @@ lowerFirst :: String -> String
 lowerFirst = \case
   [] -> []
   (x : xs) -> toLower x : xs
+
+tyVarBndrToName :: TyVarBndr flag -> Name
+tyVarBndrToName = \case
+  PlainTV name _ -> name
+  KindedTV name _ _ -> name
```

Comment:
I have no idea what nuance exists here. I imagine most of the applications will be simple types like the one represented in the test.

Reply:
It's a type variable with a kind annotation:

```
data OtherType a b (kinded :: *  -> *) (kindedSymbol :: Symbol) = OtherType
  { otherTypeField :: Maybe a
  , otherTypeOtherField :: Either a b
  , otherTypeKindedConstructor :: kinded ()
  , otherTypeKindedSymbol :: Proxy kindedSymbol
  }
```

Reply:
The implementation probably works with the kinded type variables too - they should be inferred.

Reply:
I found that `KindedTV` is generated even for type parameters without annotations. Like when I was hacking on some TH code, I kept getting errors about `Foo (a :: Type) (b :: Type)` when I wrote the type as `data Foo a b = ...`.

Reply:
If I added `kindedSymbol` to `OtherType` the test it still works. I'll create an issue to add these to the test.